### ISSUE TITLE
002 feedback

### ISSUE THREAD

**Issue by daniel-lo** [Causch/ioBroker.network#2]
002 makes objects .. nice .. but ..
- Presence notification doesn't work .. devices I can ping show "0"
- I think it would be good to have a means to resolve the names .. maybe upload a host file .. or even only edit the names in a table (which can be saved and reused)

**Comment by Causch** [comment]
DNS names are in in the 0.0.3b release, also fixed some stuff regarding arping and rescan of devices.

**Comment by daniel-lo** [comment]
Everything is wonderful (incl. name resolution) .. except for the dns branch .. (unless it should be that way).
![dns](https://user-images.githubusercontent.com/4311558/48073569-541ddd00-e1df-11e8-9174-d826baf44048.png)

**Comment by daniel-lo** [comment]
Number of pakets to send?

I see there are options for WOL: wol wake on lan ( 0 = off, 1 = send one packet, 2 = interval sending )

What is the interval? And why would I control the number of pakets do send (default?)?

**Comment by daniel-lo** [comment]
Could you also add the alive indicators in the DNS branch? It seems more comprehensive to me than the MAC list .. maybe it should include all the same flags .. just from the dns perspective.

**Comment by Causch** [comment]
Points in the name divides the tree into new leaves, which is always the case in domain names, and it looks like there is no string escaping for that in iobroker, or i didnt found one yet

**Comment by daniel-lo** [comment]
Not worried about that .. but I had no entries .. I edited above. Node tree is okay for me.

**Comment by daniel-lo** [comment]
network.0 | 2018-11-07 16:13:54.369 | warn | Error searching for reverse entry
-- | -- | -- | --
network.0 | 2018-11-07 16:13:54.368 | warn | Error searching for reverse entry
network.0 | 2018-11-07 16:13:54.368 | warn | Error searching for reverse entry
network.0 | 2018-11-07 16:13:54.368 | warn | Error searching for reverse entry
network.0 | 2018-11-07 16:13:54.368 | warn | Error searching for reverse entry
network.0 | 2018-11-07 16:13:54.367 | warn | Error searching for reverse entry
network.0 | 2018-11-07 16:13:54.367 | warn | Error searching for reverse entry

**Comment by Causch** [comment]
you get this error, when you dont have a reverse entry for that ip in your nameserver. If you have an own NS, you should create reverse Zone for your range. In my case it's 111.168.192.in-addr.arpa. On Fritz.Box i believe its enough to give a name for that IP and just save it

**Comment by daniel-lo** [comment]
maybe it makes sense to restrict logs .. for certain situations .. I personally would suppress these errors and also the information every second that the alive query has run. Probably a matter of taste ;-)

**Comment by Causch** [comment]
Yes, i will change the log level on all the entries, so that just real errors are shown

**Comment by daniel-lo** [comment]
Looking forward to your next build .. and to see if you adapt the suggestions above :-)

**Comment by daniel-lo** [comment]
Deleted dns node .. but adapter doesn't seem to like that ... stays red .. in other cases the objects are just rebuilt. Adapter is gone and seemingly can't be reinstalled:

`
host.ioBroker | 2018-11-11 06:05:15.807 | error | instance system.adapter.network.0 terminated with code 1 ()
-- | -- | -- | --
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at startup (bootstrap_node.js:204:16)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at Function.Module.runMain (module.js:694:10)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at Function.Module._load (module.js:498:3)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at tryModuleLoad (module.js:506:12)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at Module.load (module.js:566:32)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at Object.Module._extensions..js (module.js:664:10)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at Module._compile (module.js:653:30)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at Object.<anonymous> (/opt/iobroker/node_modules/iobroker.network/main.js:15:17)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at new Adapter (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:160:32)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:     at JSON.parse (<anonymous>)
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]: SyntaxError: Unexpected token } in JSON at position 685
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:                                                                 ^
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]:                 "de": "fixed arp-scan command line",            },
host.ioBroker | 2018-11-11 06:05:15.806 | error | Caught by controller[0]: undefined:16
`

**Comment by daniel-lo** [comment]
BTW "Host/host" could be deleted for legibility .. in the object name field.

**Comment by daniel-lo** [comment]
Restarted ioBroker .. but can't reinstall .. strange

**Comment by daniel-lo** [comment]
Works thanks .. detect presence on/of = false ???? Objects aren't fully in sync yet .. as far as I notice. Not all options in instance settings are clear to me .. or better said could be interpreted in multiple ways.

Please let me know if I give too much feedback :-)

**Comment by daniel-lo** [comment]
Hmm .. my vacuum cleaner is still present .. even though it is on its way back to the vendor .. strange ;-)

**Comment by daniel-lo** [comment]
Just made this screenshot .. device has been away for a while. Strange values all-in-all:

![wert](https://user-images.githubusercontent.com/4311558/48669672-b04efe00-eb09-11e8-8cfc-563171dfd3c6.png)

**Comment by Causch** [comment]
Did you put detect on true? Otherwise it will set up the default value to true

**Comment by daniel-lo** [comment]
you mean the option that was unclear to me (see above).

"Detect presence on/off" is set to false .. whatever that means .. I didn't change any options and my expectation is that a network scanner gives me a true when a device is present - otherwise false.

"Detect presence on/off" in my interpretation would mean that a defide is exluded if "Detect presence" is set to false. If this is corrent then "on/off" is confusing and why would this default to "false"???

**Comment by Causch** [comment]
sorry, i was busy in last couple of days.
Detect presence is default off, and should be turned on for devices, where you really want to detect the presence ( e.g. mobile ), as its a costly routine ( can sometimes take 3 retries of arping, or wakeup for iphone before arping )
The scanner ( full scan ) is not 100% suitable for presence detection, as some devices dont respond in a timeframe, its also possible, that another process receives the arp-response, and its slow. But, as you mentioned, a missing device is not set to "not present" after a while, this will be in the next release.

**Comment by daniel-lo** [comment]
and maybe remove on/off from the string .. since the values are true/false .. no need for this information.
another idea would be to add a state disabled to the presence value .. in case presence detection is off .. if adding a state is possible

**Comment by daniel-lo** [comment]
project still alive?

**Comment by daniel-lo** [comment]
nice

**Comment by daniel-lo** [comment]
In case the statement is still valid .. mDNS may be a good option.

**Comment by EYRudakov** [comment]
Good afternoon, for some reason, he writes:
"network.0 | 2021-02-08 13:31:20.553 | error | (10204) Error running arp 1 ERROR: Cannot open raw packet socket socket: Operation not permitted"
